internalshared/configutil/telemetry_test.go

```diff
@@ -0,0 +1,52 @@
+package configutil
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParsePrefixFilters(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		inputFilters            []string
+		expectedErrStr          string
+		expectedAllowedPrefixes []string
+		expectedBlockedPrefixes []string
+	}{
+		{
+			[]string{""},
+			"Cannot have empty filter rule in prefix_filter",
+			[]string(nil),
+			[]string(nil),
+		},
+		{
+			[]string{"vault.abc"},
+			"Filter rule must begin with either '+' or '-': \"vault.abc\"",
+			[]string(nil),
+			[]string(nil),
+		},
+		{
+			[]string{"+vault.abc", "-vault.bcd"},
+			"",
+			[]string{"vault.abc"},
+			[]string{"vault.bcd"},
+		},
+	}
+	t.Run("validate metric filter configs", func(t *testing.T) {
+		t.Parallel()
+
+		for _, tc := range cases {
+
+			allowedPrefixes, blockedPrefixes, err := parsePrefixFilter(tc.inputFilters)
+
+			if err != nil {
```

Comment:
It's fine with the cases you have now, but I consider it best practice when you have an expected error field in your test case type to do something like

if err != nil { compare with expected error }
else { assert expected error empty}

Reply:
ah that makes sense! added that here 6c2ad99